File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        l = 0
        r = n - 1

        copy = nums.copy()
        copy.sort()  # sort array for quick search O(n)
        reverseCopy = nums.copy()  # copy array for search O(n)
        reverseCopy.reverse()

        while l < r:
            sumTwo = copy[l] + copy[r]

            if sumTwo == target:
                result = [nums.index(copy[l]), n - 1 - reverseCopy.index(copy[r])]
                result.sort()
                return result

            if sumTwo < target:
                # if(target is greater than sum), move start to the next index
                l += 1
            elif sumTwo > target:
                # if(target is smaller than sum), move start to the prev index
                r -= 1

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const n = nums.length;
    let l = 0;
    let r = n - 1;

    // Create copies of the array
    const copy = [...nums].sort((a, b) => a - b); // sort array for quick search O(n)
    const reverseCopy = [...nums].reverse(); // copy array for search O(n)

    while (l < r) {
        const sumTwo = copy[l] + copy[r];

        if (sumTwo === target) {
        const result = [
            nums.indexOf(copy[l]),
            n - 1 - reverseCopy.indexOf(copy[r]),
        ];
        result.sort((a, b) => a - b);
        return result;
        }

        if (sumTwo < target) {
        // if(target is greater than sum), move start to the next index
        l++;
        } else if (sumTwo > target) {
        // if(target is smaller than sum), move start to the prev index
        r--;
        }
    }

    // In case no solution is found
    return [];

};